cliconfig/config.go

```diff
@@ -29,20 +31,15 @@ var (
 	configDir = os.Getenv("DOCKER_CONFIG")
 )
 
-func getDefaultConfigDir(confFile string) string {
-	confDir := filepath.Join(homedir.Get(), confFile)
-	// if the directory doesn't exist, maybe we called docker with sudo
-	if _, err := os.Stat(configDir); err != nil {
-		if os.IsNotExist(err) {
-			return filepath.Join(homedir.GetWithSudoUser(), confFile)
-		}
-	}
-	return confDir
-}
-
 func init() {
 	if configDir == "" {
-		configDir = getDefaultConfigDir(".docker")
+		configDir = filepath.Join(homedir.Get(), defaultConfigDir)
+		if _, err := os.Stat(configDir); err != nil {
+			// if the directory doesn't exist, maybe we called docker with sudo
```

Comment:
I don't think that's true. If neither root nor user has a .docker dir then configDir will end up being the user's not root's. So the default isn't root, its ~user.

But I don't understand why we're falling back to to the user's at all. I believe the code will do this:
- look at ~root/.docker
- if not found, use ~user/.docker
- any updates will be written to ~user/.docker
But, if ~root/.docker exists then we'll save any udpated to ~root/.docker.
So the location of where we store things (root vs sudo user) will change based on whether something already exists or not - which mean the sudo user's config file will be updated inconsistently based on whether root has a config file or not.  I don't think this randomness would be expected.  If I'm running something as root I expect all my updates to be done as root - including ownership.  So, who will own the config file? root?  Seems odd to be in ~user's dir then.
I'm so baffled by this  :-)
Also, back to my other question... do other tools act this way?